src/libasr/codegen/llvm_utils.cpp

```diff
@@ -1601,6 +1601,10 @@ namespace LCompilers {
         builder->CreateBr(mergeBB);
         llvm_utils->start_new_block(elseBB);
         {
+            if (def_value != nullptr) {
+                llvm_utils->start_new_block(mergeBB);
+                return def_value;
+            }
```

Comment:
Shouldn't this be placed where we are raising the error like if `def_value` is not `nullptr` then generate code for using it, if it is `nullptr` then generate code for raising error. As of now I see that it will always return and never the generate code for reading the item if `def_value` is not `nullptr`.